core/src/test/java/com/linecorp/armeria/server/management/JvmManagementServiceTest.java

```diff
@@ -0,0 +1,105 @@
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.ResponseHeaders;
+import com.linecorp.armeria.common.SplitHttpResponse;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.testing.junit5.server.ServerExtension;
+
+import reactor.core.publisher.Flux;
+
+class JvmManagementServiceTest {
+    private static final ObjectMapper mapper = new ObjectMapper();
+
+    @RegisterExtension
+    static ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) {
+            sb.serviceUnder("/internal/management", JvmManagementService.of());
+        }
+    };
+
+    @Test
+    void threadDump() throws InterruptedException {
+        final WebClient client = WebClient.of(server.httpUri());
+        final AggregatedHttpResponse response =
+                client.get("/internal/management/threaddump").aggregate().join();
+        assertThat(response.contentType()).isEqualTo(MediaType.PLAIN_TEXT);
+        assertThat(response.contentUtf8()).contains(Thread.currentThread().getName());
+    }
+
+    @Test
+    void threadDumpWithJson() throws Exception {
+        final WebClient client = WebClient.of(server.httpUri());
+        final AggregatedHttpResponse response =
+                client.prepare()
+                      .get("/internal/management/threaddump")
+                      .header(HttpHeaderNames.ACCEPT, MediaType.JSON)
+                      .execute().aggregate().join();
+
+        assertThat(response.contentType()).isEqualTo(MediaType.JSON);
+        final String content = response.contentUtf8();
+        assertThat(content).contains(Thread.currentThread().getName());
+        assertThat(mapper.readTree(content).isArray()).isTrue();
+    }
+
+    @Test
+    void heapDump() throws InterruptedException {
+        final WebClient client = WebClient.of(server.httpUri());
+        final HttpResponse response = client.get("/internal/management/heapdump");
+        final SplitHttpResponse splitHttpResponse = response.split();
+        final ResponseHeaders headers = splitHttpResponse.headers().join();
+        final ContentDisposition disposition =
+                ContentDisposition.parse(headers.get(HttpHeaderNames.CONTENT_DISPOSITION));
+
+        assertThat(disposition).isNotNull();
+        final String filename = disposition.filename();
+        assertThat(filename).startsWith("heapdump_");
+        assertThat(filename).endsWith(".hprof");
+
+        final byte[] fileHeader = "JAVA PROFILE".getBytes(StandardCharsets.UTF_8);
+        final AtomicInteger counter = new AtomicInteger();
+        final byte[] actual = Flux.from(splitHttpResponse.body())
+                                  .map(HttpData::array)
+                                  .takeUntil(bytes -> counter.getAndAdd(bytes.length) <= fileHeader.length)
+                                  .reduce(Bytes::concat)
+                                  .block();
```

Comment:
Only if we had these methods in `StreamMessage` :laughing:

Reply:
Indeed! 😆